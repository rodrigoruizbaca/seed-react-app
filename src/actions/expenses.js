import uuid from 'uuid';
import moment from 'moment';

export const addExpense = ({
    description = '',
    note = '',
    amount = 0,
    createdAt = moment()
    }) => {
        return {
            type: 'ADD_EXPENSE',
            expense: {
                id: uuid(),
                description,
                note,
                amount,
                createdAt
              }
        };
};