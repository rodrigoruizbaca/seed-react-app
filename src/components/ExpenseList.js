import React from 'react';
import {connect} from 'react-redux';
import ExpenseItem from './ExpenseListItem';
import expenseSelector from '../selectors/expenses';

export const ExpenseList = (props) => {
    console.log(props);
    return (
        <div>
            {
                props.expenses.map((expense) => {
                    return <ExpenseItem key={expense.id} expense={expense}/>
                })
            }
        </div>
    );
};

const mapStateToProps = (state) => {
    return {'expenses': expenseSelector(state.expenses, state.filters)};
};


export default connect(mapStateToProps)(ExpenseList);
