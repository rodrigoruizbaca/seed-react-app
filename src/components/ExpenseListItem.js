import React from 'react';

const ExpenseItem = (props) => (
    <div>
        <h3>{props.expense.description}</h3>
        <p>{props.expense.amount}-{props.expense.createdAt}</p>
    </div>
);

export default ExpenseItem;