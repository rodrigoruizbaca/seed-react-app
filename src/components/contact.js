import React from 'react';
import ReactDOM, {NavLink} from 'react-dom';

export default Contact => (
    <div>
        Contact
    </div>
);