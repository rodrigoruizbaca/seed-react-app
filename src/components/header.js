import React from 'react';
import ReactDOM from 'react-dom';
import {NavLink} from 'react-router-dom';

export default Header => (
    <div>
        <h1>Porftolio</h1>
        <NavLink to="/">Home</NavLink>
        <NavLink to="portfolio/123">Porftolio</NavLink>
        <NavLink to="contact">Contact</NavLink>
    </div>
);
