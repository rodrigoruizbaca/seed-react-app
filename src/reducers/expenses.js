
const expensesReducer = (state = [], action) => {
    switch (action.type) {
        case 'GET_EXPENSES': {
            return state;
        }
        case 'SET_EXPENSES': {
            const expenses = action.expenses;
            return expenses;
        }
        case 'ADD_EXPENSE': {
            const expense = action.expense;
            return [...state, expense];
        }
        default:
            return state;
    }
};

export default expensesReducer;