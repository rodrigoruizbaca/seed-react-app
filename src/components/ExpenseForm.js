import React from 'react';
import {addExpense} from '../actions/expenses';
import {connect} from 'react-redux';
import moment from 'moment';

export class ExpenseFormComponent extends React.Component {



    state =  {
        description: '',
        amount: 0,
        createdAt: moment()
    };

    onDescriptionChange = (e) => {
        const description = e.target.value;
        this.setState(() => {
            return {description};
        });
    };    

    onAmountChange = (e) => {
        const amount = e.target.value;
        this.setState(() => {
            return {amount};
        });
    };       

    handleSubmit = (e) => {
        e.preventDefault();
        const expense = {description: this.state.description, amount: this.state.amount};
        this.props.dispatch(addExpense(expense));
    };

    render() {
        return (
            <div>
                <form onSubmit={this.handleSubmit}>
                    <label>
                        Description:
                        <input type="text" autoFocus value={this.state.description} onChange={this.onDescriptionChange}/>
                    </label>
                    <label>
                        Amount:
                        <input type="number" value={this.state.amount} onChange={this.onAmountChange}/>
                    </label>
                    <label>
                        createdAt:
                        <input type="number" value={this.state.createdAt}/>
                    </label>
                    <input type="submit" value="Submit" />
                </form>
            </div>
        );
    }
}

export default connect()(ExpenseFormComponent);
