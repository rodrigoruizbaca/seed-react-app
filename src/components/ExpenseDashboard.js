import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpenseForm from './ExpenseForm';

const ExpenseDashboard = () => {
    return ( 
        <div>
            <ExpenseForm/>
            <p/>
            <ExpenseListFilters/>
            <ExpenseList/>
        </div>
    );
};

export default ExpenseDashboard;