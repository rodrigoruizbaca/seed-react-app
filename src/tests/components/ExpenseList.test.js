import React from 'react';
import {shallow} from 'enzyme';
import {ExpenseList} from '../../components/ExpenseList';
import expenses from '../fixtures/expenses';


test('should create three items', () => {
    const wrapper = shallow(<ExpenseList expenses={expenses}/>);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('ExpenseItem').length).toBe(3);
});