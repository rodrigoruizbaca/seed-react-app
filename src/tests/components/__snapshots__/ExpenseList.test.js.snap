// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create three items 1`] = `
<div>
  <ExpenseItem
    expense={
      Object {
        "amount": 195,
        "createdAt": 0,
        "description": "Gum",
        "id": "1",
        "note": "",
      }
    }
    key="1"
  />
  <ExpenseItem
    expense={
      Object {
        "amount": 109500,
        "createdAt": -345600000,
        "description": "Rent",
        "id": "2",
        "note": "",
      }
    }
    key="2"
  />
  <ExpenseItem
    expense={
      Object {
        "amount": 4500,
        "createdAt": 345600000,
        "description": "Credit Card",
        "id": "3",
        "note": "",
      }
    }
    key="3"
  />
</div>
`;
